src/compiler/checker.ts

```diff
@@ -5057,11 +5057,11 @@ namespace ts {
             return type.resolvedBaseTypes;
         }
 
-        function resolveBaseTypesOfClass(type: InterfaceType): void {
-            type.resolvedBaseTypes = emptyArray;
+        function resolveBaseTypesOfClass(type: InterfaceType): BaseType[] {

```

Comment:
It doesn't look like you are using this return type.

Reply:
No, I was just using it to ensure I set `resolvedBaseType` at each exit point. I can replace it with `void` if you'd like.